### PR TITLE
fix: restore hidden wallet fallback selection(OK-57468)

### PR DESCRIPTION
**Pull request by originalix**
[OK-57468](https://onekeyhq.atlassian.net/browse/OK-57468)

----------

## Summary
- Keep an available standard HD/HW/QR wallet selected when the previously selected hidden wallet is no longer accessible.
- Add regression coverage for the single hardware wallet + hidden wallet case where the standard wallet has no indexed account yet.

## Intent & Context
A Slack thread reported that after adding a hidden passphrase wallet in OneKey App, disabling "Keep accessible", restarting the app, and returning to the wallet page, the page becomes blank. QA reproduced the issue when the user only has one hardware wallet plus a passphrase wallet. The related Jira is OK-57468.

The expected behavior is to select the standard wallet if possible, or show an empty/create state instead of leaving the wallet page blank.

## Root Cause
When a hidden wallet is not kept accessible, it becomes a temporary wallet. After app restart, the background runtime's process-local `tempWallets` state is empty, so the hidden wallet is treated as removed and filtered out of visible wallet lists.

`autoSelectNextAccount` previously only selected HD/HW/QR wallets that already had indexed accounts. In the single hardware wallet scenario, the standard hardware wallet can still exist but have no indexed account, so the fallback loop skipped it and cleared `walletId` / `focusedWallet`. Home then had no active wallet/account state to render, while the wallet list was not necessarily a true no-wallet list, producing the blank wallet page.

## Design Decisions
The fix stays in account selector fallback logic instead of adding a Home-only visual workaround. When no HD/HW/QR wallet with indexed accounts is found, the selector now keeps the first non-deprecated/non-mocked HD/HW/QR wallet selected as an empty account wallet. This lets existing Home empty-account/create-address UI handle the state naturally.

This preserves the existing priority: wallets with indexed accounts are still preferred. The new fallback only runs after no indexed-account wallet is available.

## Changes Detail
- `packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx`: track the first available HD/HW/QR wallet during auto-select and keep it selected when no wallet with indexed accounts exists.
- `packages/kit/src/states/jotai/contexts/accountSelector/actions.test.tsx`: add a regression test for an unavailable hidden wallet falling back to an empty standard hardware wallet.

## Risk Assessment
- **Risk Level**: Medium
- **Affected Platforms**: Mobile, Desktop, Extension, Web surfaces that use account selector; the reported reproduction is Android/iOS.
- **Risk Areas**: Account selector fallback behavior after wallet/account removal or hidden wallet temp-state filtering. The change is scoped to the no-indexed-account fallback path and preserves existing selection priority for wallets with accounts.

## Test plan
- [x] `PATH=/Users/leon/.cache/codex-runtimes/codex-primary-runtime/dependencies/node/bin:$PATH yarn jest packages/kit/src/states/jotai/contexts/accountSelector/actions.test.tsx --runInBand`
- [x] `PATH=/Users/leon/.cache/codex-runtimes/codex-primary-runtime/dependencies/node/bin:$PATH npx oxlint --tsconfig ./tsconfig.json --type-aware packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx packages/kit/src/states/jotai/contexts/accountSelector/actions.test.tsx --deny-warnings`
- [x] `PATH=/Users/leon/.cache/codex-runtimes/codex-primary-runtime/dependencies/node/bin:$PATH yarn tsc:staged`
- [x] `PATH=/Users/leon/.cache/codex-runtimes/codex-primary-runtime/dependencies/node/bin:$PATH yarn lint:staged`

[OK-57468]: https://onekeyhq.atlassian.net/browse/OK-57468?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 1]
```diff
@@ -2911,19 +2911,30 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
...
+                    })
+                  ) {
+                    selectedWallet = wallet0;
+                    selectedWalletId = selectedWallet?.id;
+                    selectedAccountNew.walletId = selectedWalletId;
+                    break;
+                  }
                 }
               }
+              if (!selectedWallet && firstAvailableWallet) {
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx]: 当当前选中的隐藏钱包在 `bg` 里只是被标成 mocked/deprecated，而 `buildActiveAccountInfoFromSelectedAccount()` 仍把这条 wallet record 回传给 `main` 时，`selectedWallet` 进入这段逻辑前就是 truthy。

这样这里的 `if (!selectedWallet && firstAvailableWallet)` 不会命中，下面的 reset 分支又会把 selection 直接清空，结果还是回不到空硬件钱包，用户也拿不到后续创建首个账户的入口。

这里建议把 mocked/deprecated wallet 也当作“不可用当前钱包”处理，再决定是否落到 `firstAvailableWallet`，并补一条 `activeAccount.wallet` 仍是 zombie wallet 记录的测试。

Comment (sidmorizon) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx]: # 🟡 P2 空钱包 fallback 未覆盖 deprecated/mocked 当前钱包

## 问题

新增的 fallback 条件是 `if (!selectedWallet && firstAvailableWallet)`，只有在 `selectedWallet` 为空时才会回退到空的可用钱包。但进入这段 `shouldSelectHdHwWallet` 逻辑的触发条件之一，正是「当前 `selectedWallet` 是 deprecated/mocked」（见上方 line 2887 以及紧接着 reset 块 line 2940-2941 的 `accountUtils.isWalletDeprecatedOrMocked(selectedWallet)`）。

此时 `selectedWallet` 仍是那个 deprecated 钱包（真值），循环若没找到含 indexed account 的钱包也不会重置它。于是 fallback 因 `!selectedWallet` 为 `false` 被跳过，随后 reset 块因 `isWalletDeprecatedOrMocked(selectedWallet)` 为真，把 `walletId` / `indexedAccountId` / `focusedWallet` 全部重置为 `undefined`。

## 影响

当用户当前聚焦在一个 deprecated/mocked 钱包、没有任何钱包含 indexed account、但存在一个非 deprecated 的空 HD/HW/QR 钱包（即 `firstAvailableWallet`）时，本该回退到该空钱包，却被重置为「无钱包」。这既让本 PR 新增的空钱包回退在 deprecated 触发路径下失效，也与本函数注释（line 2838-2840）「deprecated/mocked 钱包应挑选下一个有效钱包或重置」的意图不一致——fallback 条件是 reset 条件的严格子集，两者不对齐。

## 建议

让 fallback 条件与 reset 条件对齐，覆盖 deprecated/mocked 当前钱包：

```ts
const needsReplacement =
  !selectedWallet || accountUtils.isWalletDeprecatedOrMocked(selectedWallet);
if (needsReplacement && firstAvailableWallet) {
  selectedWallet = firstAvailableWallet;
  // ...
}
```

同一个 `needsReplacement` 也可复用到下方 reset 判断，避免以后再次漂移。

[Claude session ↗](https://claude.ai/code/session_01KxUYvTprovz6C2owCgYFCm)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 2]
```diff
@@ -3264,19 +3264,33 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
...
+                  ) {
+                    selectedWallet = wallet0;
+                    selectedWalletId = selectedWallet?.id;
+                    selectedAccountNew.walletId = selectedWalletId;
+                    break;
+                  }
                 }
               }
+              if (
+                (!selectedWallet || !hasIndexedAccounts) &&
```
Comment (sidmorizon) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx]: # [🔴 P1] 回退守卫复用循环前的 `hasIndexedAccounts`，会覆盖循环已选中的有账户钱包

## 问题

`hasIndexedAccounts`（第 3224 行）是在 `for` 循环之前基于**最初**选中的 `selectedWalletId` 计算的 `const`，全程不重算。循环（3268-3282）一旦找到含 indexed account 的钱包就 `break` 并把 `selectedWallet = wallet0`（有账户的钱包）。但本行守卫里的 `hasIndexedAccounts` 仍是原始值：只要最初钱包没有 indexed accounts（未选中 / 空钱包 / deprecated，也正是进入本分支的常见前提），`!hasIndexedAccounts` 恒为 `true`。

同时 `firstAvailableWallet`（第 3270 行）是循环里**第一个**非 deprecated 钱包，可能是空钱包；当空钱包排在有账户钱包之前时，它与 `break` 选中的 `wallet0` 并非同一个：

```ts
firstAvailableWallet = firstAvailableWallet || wallet0; // 先记下第一个非 deprecated（可能是空钱包）
if (await isWalletHasIndexedAccounts({ walletId: wallet0.id })) {
  selectedWallet = wallet0; break;                       // 后面才 break 到有账户的钱包
}
```

## 影响

场景:钱包列表为 `[空的标准 HW 钱包, 有账户的隐藏 HW 钱包]`。循环本已正确 `break` 选中有账户的隐藏钱包,但本守卫变成 `(false || true) && firstAvailableWallet` = `true`,于是用 `firstAvailableWallet`(空的标准钱包)覆盖它,并置 `indexedAccountId = undefined`。第 3325 行的恢复逻辑基于 `selectedWalletId`(=空钱包)`getIndexedAccountsOfWallet` 取到 `[]`,无法补回,用户最终落到**空钱包**。这破坏了本分支"优先选择有 indexed accounts 的钱包"的既定优先级。相对 `bd72d4c` 的旧条件 `!selectedWallet`(循环命中后为 `false`,不会误触发)这是**新引入的回归**,现有新增测试(单一钱包、`isWalletHasIndexedAccounts` 恒 `false`)覆盖不到该路径。

## 建议

不要复用循环前的 `hasIndexedAccounts`,改用循环内的命中标志来区分"循环已找到有账户钱包"与"循环没找到、当前钱包不可用":

```ts
let foundHdHwWalletWithIndexedAccounts = false;
for (const wallet0 of wallets) {
  if (!accountUtils.isWalletDeprecatedOrMocked(wallet0)) {
    firstAvailableWallet = firstAvailableWallet || wallet0;
    if (await serviceAccount.isWalletHasIndexedAccounts({ walletId: wallet0.id })) {
      selectedWallet = wallet0;
      selectedWalletId = selectedWallet?.id;
      selectedAccountNew.walletId = selectedWalletId;
      foundHdHwWalletWithIndexedAccounts = true;
      break;
    }
  }
}
if ((!selectedWallet || !foundHdHwWalletWithIndexedAccounts) && firstAvailableWallet) {
  // ...
}
```

这样循环命中时不再被覆盖,循环未命中(当前钱包空 / deprecated)时仍能回退到 `firstAvailableWallet`,原修复意图保留。建议补一个"空钱包在前、有账户钱包在后"的多钱包测试。

[Claude session ↗](https://claude.ai/code/session_01KxUYvTprovz6C2owCgYFCm)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.test.tsx [hunk 3]
```diff
@@ -1105,4 +1122,59 @@ describe('useAccountSelectorActions', () => {
...
+      focusedWallet: 'hw-standard--hidden',
+    };
+    const standardWallet = {
+      id: 'hw-standard',
+      name: 'Standard wallet',
+    } as IWallet;
+
+    mockGetAllHdHwQrWallets.mockResolvedValue({
+      wallets: [standardWallet],
+    });
```
Comment (sidmorizon) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.test.tsx:1213]: **Test only covers single-wallet scenario, missing the multi-wallet edge case**

Type: `Flag`

The new test at line 1126 ("selects an empty hardware wallet after a hidden wallet becomes unavailable") sets up `mockGetAllHdHwQrWallets` with only a single wallet (`[standardWallet]`). In this case, `firstAvailableWallet` and the loop's result are effectively the same wallet, so the fallback overwrite is harmless. A test with multiple wallets (e.g., `[emptyHwWallet, hdWalletWithAccounts]`) would expose the bug where the fallback incorrectly overwrites the valid selection from the loop.

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.test.tsx [hunk 4]
```diff
@@ -1105,4 +1122,59 @@ describe('useAccountSelectorActions', () => {
...
+    store.set(accountSelectorStorageInitDoneAtom(), true);
+    store.set(activeAccountsAtom(), {
+      0: {
+        ...defaultActiveAccountInfo(),
+        ready: true,
+        network: {
+          id: 'onekeyall',
+        } as ReturnType<typeof defaultActiveAccountInfo>['network'],
+      },
+    });
```
Comment (sidmorizon) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.test.tsx:1231]: **Test does not exercise the deprecated-wallet code path it was designed to cover**

Type: `Flag`

The new test at `actions.test.tsx:300-353` sets up a hidden wallet scenario but uses `defaultActiveAccountInfo()` which has `wallet: undefined`. This means the test exercises the `!wallet` branch of `shouldAutoSelectNextAccount` (line `actions.tsx:2844`), not the `accountUtils.isWalletDeprecatedOrMocked(wallet)` branch (line `actions.tsx:2845`). In production, a deprecated hardware wallet would still have a non-null wallet object with `isMocked: true` or `deprecated: true`. The test should set the `wallet` field in the active account to a deprecated wallet object (e.g., `{ ...standardWallet, isMocked: true }`) to properly cover the intended scenario.

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx (hunk 2 above)
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx]: 当当前选中的隐藏/硬件钱包没有 `indexedAccount` 时，这里的 fallback 条件仍然沿用进入循环前的 `hasIndexedAccounts`。如果 `wallets` 里前面是空钱包、后面才是有 `indexedAccount` 的钱包，上面的循环虽然已经命中了后者，但这里仍会再次进入 fallback，把选择覆盖回 `firstAvailableWallet`。

结果是 auto-select 会把用户从本来可以直接恢复的可用钱包切回空钱包，隐藏钱包失效后的恢复优先级也被打乱。

建议把 fallback 改成只在“循环结束后仍未选到带 `indexedAccount` 的钱包”时才触发，最好用一个独立标记记录是否命中过可用钱包，并补一个 `[emptyWallet, walletWithIndexedAccount]` 的回归用例。

Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx]: 这里的 fallback 条件仍然复用了外层旧钱包的 `hasIndexedAccounts`。

触发条件是当前 hidden wallet 没有 indexed account，但 `getAllHdHwQrWallets()` 扫描到后面的某个钱包其实已经有 indexed account。循环里虽然已经把 `selectedWallet` 切到了那个可用钱包，`!hasIndexedAccounts` 还是会保持为 `true`，于是这里又把结果覆盖回 `firstAvailableWallet`。

结果是用户会从一个可直接进入的 wallet 被错误带到第一个空 wallet / QR wallet，首页看起来像资产丢失。

建议改成“只有整轮扫描都没找到带 indexed account 的钱包时，才回退到 `firstAvailableWallet`”，单独维护一个本轮扫描结果标记，不要复用旧钱包的 `hasIndexedAccounts`。

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 5]
```diff
@@ -3264,19 +3270,38 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
...
+                firstAvailableWallet
+              ) {
+                selectedWallet = firstAvailableWallet;
+                selectedWalletId = selectedWallet.id;
+                selectedAccountNew.walletId = selectedWalletId;
+                selectedAccountNew.indexedAccountId = undefined;
+                selectedAccountNew.othersWalletAccountId = undefined;
+                selectedAccountNew.focusedWallet = selectedWalletId;
+              }
```
Comment (sidmorizon) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3565]: **Wallet auto-selection can pick a wallet without accounts over one that has accounts**

Severity: `non-severe`

The fallback condition uses a stale check (`!hasIndexedAccounts` at `packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3295`) computed from the *original* wallet, not the wallet just found by the search loop, so when the loop successfully finds a wallet with accounts but the original wallet had none, the fallback overwrites the good selection with the first available wallet (which may lack accounts).

Impact: Users with multiple hardware or HD wallets may land on an empty wallet instead of one that has accounts ready to use.

<details>
<summary>Stale const causes incorrect overwrite after successful loop selection</summary>

`hasIndexedAccounts` is a `const` computed at `packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3230-3240` based on the original `selectedWalletId`. When the original wallet is cleared (e.g. a temp wallet is removed at line 3223-3224), `hasIndexedAccounts` is `false`.

The for-loop at lines 3274-3292 iterates wallets and may find one with indexed accounts, setting `selectedWallet = wallet0` at line 3287 and breaking. After the loop, the condition at line 3294-3296:

```
if (
  (!selectedWallet || !hasIndexedAccounts) &&
  firstAvailableWallet
)
```

evaluates `!hasIndexedAccounts` as `true` (stale value from the original wallet), making the whole condition true even though `selectedWallet` was just set to a valid wallet with accounts. The code at lines 3298-3303 then overwrites `selectedWallet` with `firstAvailableWallet`, which may be an earlier wallet in the list that has no indexed accounts.

The existing test cases don't catch this because they only use a single wallet in `mockGetAllHdHwQrWallets`, so `firstAvailableWallet` and the loop-selected wallet are always the same object.

The fix should either track whether the loop successfully found a wallet with indexed accounts (e.g. a boolean flag), or check the newly-selected wallet's indexed account status instead of the stale `hasIndexedAccounts` const.
</details>

Comment (sidmorizon) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3565]: **New fallback to empty wallets is a significant behavioral change**

Type: `Flag`

Before this PR, the auto-select loop at `packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3268-3325` would only select a wallet if it had indexed accounts. The new `firstAvailableWallet` fallback (lines 3273, 3294-3304) allows selecting a wallet with zero indexed accounts, resulting in a selection with `indexedAccountId: undefined`. This is intentional for the hidden-wallet-removed scenario, but it changes the contract: downstream code that previously could assume `selectedWallet` implies `indexedAccountId` may now encounter a wallet selection without an indexed account. Verify that all consumers of the selected account handle this new state gracefully.

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 6]
```diff
@@ -3264,19 +3270,38 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
...
+                  ) {
+                    selectedWallet = wallet0;
+                    selectedWalletId = selectedWallet?.id;
+                    selectedAccountNew.walletId = selectedWalletId;
+                    break;
+                  }
                 }
               }
+              if (
+                (!selectedWallet || !hasIndexedAccounts) &&
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx]: 当当前隐藏钱包被移除、`wallets` 列表里前面是空钱包、后面还有已经创建过账户的钱包时，这里的兜底条件会误判。`hasIndexedAccounts` 是进入循环前基于旧选中钱包算出来的；即使上面的遍历已经找到后面的 `wallet0` 并把 `selectedWallet` 切过去了，只要旧值是 `false`，这里仍然会进入 fallback，把结果重新覆盖成 `firstAvailableWallet`。

这样会把本来应该恢复到“已有账户的钱包”的场景降级成“空钱包”，用户继续看到空账户内容，实际没有修复多钱包恢复路径。

建议把兜底条件改成只有在循环结束后“仍未找到带 indexed accounts 的可用钱包”时才回退，例如单独维护一个 `foundWalletWithIndexedAccounts` 标记，或者基于最终 `selectedWalletId` 重新判断，而不要复用进入循环前的 `hasIndexedAccounts`。

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 7]
```diff
@@ -3264,19 +3349,40 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
...
-                  selectedWalletId = selectedWallet?.id;
-                  selectedAccountNew.walletId = selectedWalletId;
-                  break;
+                const isWalletUnavailable =
+                  accountUtils.isWalletDeprecatedOrMocked(wallet0) ||
+                  (await serviceAccount.isTempWalletRemoved({
+                    wallet: wallet0,
+                  }));
+                if (!isWalletUnavailable) {
+                  firstAvailableWallet = firstAvailableWallet || wallet0;
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3541]: 当用户移除某个 hidden wallet，而且列表里同时存在多个没有 indexed account 的 HD/HW/QR wallet 时，这里会先记住 `walletOrder` 最靠前的可用钱包，然后在找不到带 account 的钱包时直接把它当成 fallback。

这样会把当前选择从原设备的 hidden wallet 跳到另一台设备或另一类钱包，Home/Swap 会直接切到另一套地址和资产上下文。建议先按 `associatedDevice` 或 hidden wallet 对应的标准 wallet 关系过滤同源候选，只在同源标准钱包不存在时才退回全局 `firstAvailableWallet`。

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 8]
```diff
@@ -3264,19 +3353,40 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
...
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3541]: 当用户移除某个 hidden wallet，而且列表里同时存在多个没有 indexed account 的 HD/HW/QR wallet 时，这里会先记住 `walletOrder` 最靠前的可用钱包，然后在找不到带 account 的钱包时直接把它当成 fallback。

这样会把当前选择从原设备的 hidden wallet 跳到另一台设备或另一类钱包，Home/Swap 会直接切到另一套地址和资产上下文。建议先按 `associatedDevice` 或 hidden wallet 对应的标准 wallet 关系过滤同源候选，只在同源标准钱包不存在时才退回全局 `firstAvailableWallet`。

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 9]
```diff
@@ -3264,19 +3529,40 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
               await timerUtils.wait(600);
               await serviceAccount.clearAccountCache();
               const { wallets } = await serviceAccount.getAllHdHwQrWallets();
+              let firstAvailableWallet: IDBWallet | undefined;
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3532]: 当被移除的是某个 hidden wallet，而且候选 HD/HW/QR wallet 都没有 indexed account 时，这里会先记住全局顺序上的第一个可用钱包，再直接把它当成 fallback。

因为没有先按原 hidden wallet 的 `associatedDevice` / standard-wallet 关系收敛，同设备的 standard wallet 还在时，结果也可能跳到另一台设备或另一类 wallet。Home/Swap 会直接切到另一套地址和资产上下文。

建议先筛出和原 hidden wallet 同 `associatedDevice` 的 standard wallet，只在同源候选不存在时再退回全局 `firstAvailableWallet`。

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 10]
```diff
@@ -2597,8 +2762,13 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
             sceneUrl,
           });
         const recentSelectionCacheSelectedAccountsMap = recentSelectionCache
-          ? await this.repairOthersWalletNetworkPairsInSelectedAccountsMap({
-              selectedAccountsMap: recentSelectionCache.selectedAccountsMap,
+          ? await this.clearUnavailableWalletSelectionsInStorage({
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:2765]: 当这里处理的是 `recentSelectionCache` 这种 cold-start snapshot，而不是 simpleDb 读出来的权威 map 时，`clearUnavailableWalletSelectionsInStorage()` 仍然会在 helper 内直接 `saveSelectedAccount()`。

后面的 `shouldKeepColdStartSelectedAccounts()` 还没判断这份 cache 是否够新，所以一个过期的 hidden-wallet 缓存也能先把仍然有效的 simpleDb selection 清成无 identity。这样这次启动即使最终继续用 DB 态，持久化本身已经被污染，下一次冷启动或单 scene 重建时还会丢原本有效的选择。

建议先做纯内存清理，等确认 cold-start cache 真的是胜出的 source 后再落库；`selectedAccountsAtom()` 那条分支也需要同样处理。

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 11]
```diff
@@ -3264,19 +3529,40 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
...
+                    foundWalletWithIndexedAccounts = true;
+                    break;
+                  }
                 }
               }
+              if (
+                (!selectedWallet || !foundWalletWithIndexedAccounts) &&
+                firstAvailableWallet
+              ) {
+                selectedWallet = firstAvailableWallet;
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3559]: 当 hidden/mock/temp-removed wallet 被移除，而这里退回的是一个没有 indexed account 的 HD/HW/QR wallet 时，后面的 `!indexedAccountId && !othersWalletAccountId` 分支还会继续执行 others-wallet auto-select。

这样一旦 imported / watching / external 里有账号，这个空 HD/HW/QR fallback 会立刻被覆盖；而当前候选本身也只是全局 `firstAvailableWallet`，没有先收敛到原 wallet 同 `associatedDevice` 的 standard wallet。Home/Swap 最终可能直接跳到另一台设备或另一类 wallet 的资产上下文。

建议在这里先按原 wallet 的 `associatedDevice` 选同源 fallback，并在选中空 HD/HW/QR fallback 后打一个 guard，阻止后续 others-wallet fallback 再覆盖它。

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 12]
```diff
@@ -859,6 +866,158 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
...
+              num,
+              selectedAccount,
+              clearedSelectedAccount,
+              walletId: unavailableWallet.walletId,
+              isMissingWallet: unavailableWallet.isMissingWallet,
+              isDeprecatedOrMocked: unavailableWallet.isDeprecatedOrMocked,
+              isTempWalletRemoved: unavailableWallet.isTempWalletRemoved,
+            },
+          );
+          await backgroundApiProxy.simpleDb.accountSelector.saveSelectedAccount(
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:1006]: 当调用方传进来的是 `recentSelectionCache`、当前 cold-start atom，或者 discover 合并后的 map 时，这个 helper 还没决定谁才是本次启动的权威 source，就已经在这里直接 `saveSelectedAccount()` 了。

这样一个过期的 hidden-wallet snapshot 也能先把 simpleDb 里仍然有效的 selection 清成无 identity。后面即使 `shouldKeepColdStartSelectedAccounts()` 最终不采用这份 snapshot，持久化本身已经被污染，下一次冷启动或单 scene 重建时还会继续丢选择。

建议把 unavailable-wallet 清理拆成“纯计算”和“显式落库”两步，只在确认胜出的 source 后再执行 `saveSelectedAccount()`。

File: packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx [hunk 13]
```diff
@@ -3437,6 +3729,13 @@ class AccountSelectorActions extends ContextJotaiActionsBase {
               : undefined,
             builder: () => selectedAccountNew,
           });
+          await this.saveClearedSelectedAccountToStorage.call(set, {
```
Comment (originalix) [inline packages/kit/src/states/jotai/contexts/accountSelector/actions.tsx:3732]: 这里补落库只覆盖当前 scene。当前 scene 因 hidden wallet 被移除而降级成无 identity selection 时，home/swap 的 peer scene 仍然只靠 `syncHomeAndSwapSelectedAccount()` 改 atom。

随后 `AccountSelectorEffects.tsx` 的 auto-save 还是会走只接受 identity selection 的 `saveToStorage()`，所以 target scene 的 simpleDb 继续保留旧 hidden wallet。recent cache 一旦失效、app 冷启动，或者 target scene 单独重建时，旧选择就会被重新恢复。

建议在 home/swap 同步路径识别 clear 或 incomplete transition，并对 target scene 也复用这个 helper 或等价的 cleared-selection 持久化。

**Comment by originalix** [comment]
Superseded by #12340 targeting x.
